enacts/tests/test_calc.py

```diff
@@ -5,6 +5,298 @@
 import data_test_calc
 
 
+def test_groupby_dekads_perfect_partition():
+    t = pd.date_range(start="2000-05-01", end="2000-06-30", freq="1D")
+    values = 1 + np.arange(t.size)
+    precip = xr.DataArray(values, coords={"T": t})
+    grouped = calc.groupby_dekads(precip)
+
+    assert (grouped.sum() == [ 55, 155, 286, 365, 465, 565 ]).all()
+
+
+def test_groupby_dekads_noon_days():
+    t = pd.date_range(start="2000-05-01T120000", end="2000-06-30T120000", freq="1D")
+    values = 1 + np.arange(t.size)
+    precip = xr.DataArray(values, coords={"T": t})
+    grouped = calc.groupby_dekads(precip)
+
+    assert (grouped.sum() == [ 55, 155, 286, 365, 465, 565 ]).all()
+
+
+def test_groupby_dekads_overlaps():
+    t = pd.date_range(start="2000-04-30", end="2000-06-29", freq="1D")
+    values = np.arange(t.size)
+    precip = xr.DataArray(values, coords={"T": t})
+    grouped = calc.groupby_dekads(precip)
+
+    assert (grouped.sum() == [ 55, 155, 286, 365, 465 ]).all()
+
+
+def test_groupby_dekads_noon_days_overlaps():
+    t = pd.date_range(start="2000-05-01T120000", end="2000-07-01T120000", freq="1D")
+    values = 1 + np.arange(t.size)
+    precip = xr.DataArray(values, coords={"T": t})
+    grouped = calc.groupby_dekads(precip)
+
+    assert (grouped.sum() == [ 55, 155, 286, 365, 465, 565 ]).all()
+
+
+def test_interval_to_point_dim():
+    t = pd.date_range(start="2000-05-01T120000", end="2000-07-01T120000", freq="1D")
+    values = 1 + np.arange(t.size)
+    precip = xr.DataArray(values, coords={"T": t})
+    grouped = calc.groupby_dekads(precip).sum()
+    mid_dim = calc.interval_to_point(grouped["T_bins"])
+    left_dim = calc.interval_to_point(grouped["T_bins"], to_point="left")
+    right_dim = calc.interval_to_point(grouped["T_bins"], to_point="right")
+
+    assert mid_dim.name.endswith("_mid")
+    assert left_dim.name.endswith("_left")
+    assert right_dim.name.endswith("_right")
+    for t in range(mid_dim.size) :
+        assert mid_dim.values[t] == mid_dim["T_bins"].values[t].mid
```

Comment:
Could you explain? I thought `T_bins` was an array of intervals, but you're comparing them to points here? And why aren't you making any assertions about left_dim and right_dim? (Also why are they all called `*_dim`, when they're not dimensions but DataArrays?)

Reply:
T_bins is an array of intervals. .mid is an attribute of a pd.Interval but only each value of T_bins is of type pd.Interval to which .mid can be applied. So both mid_dim and T_bins... .mid are points and I am testing that my function does the right point conversion. I could do the same test for left and right... I guess I figured if it works for one attribute...

I guess I called them _dim by usage extension... since all of this is to convert coordinates (or dims... the distinction between coords and dims in xarray API/doc is sometimes confusing too). So... lots of confusion.

Reply:
Indeed, I seem to have missed the `.mid` at the end of the line.

In xarray, a coordinate is a DataArray that contains the labels for a dimension. A dimension isn't represented by any xarray class, it's just a description of how the data in an array are shaped.